air/src/execution_step/boxed_value/jvaluable/iterable_item.rs

```diff
@@ -55,7 +56,10 @@ impl<'ctx> JValuable for IterableItem<'ctx> {
         };
 
         let selected_value = select_from_scalar(jvalue, lambda.iter(), exec_ctx)?;
-        Ok((selected_value, tetraplet.clone()))
+        let mut tetraplet = tetraplet.as_ref().clone();
+        tetraplet.add_lambda(&format_ast(lambda));
```

Comment:
It was an another bug, yes

Reply:
But prob bug isn't a right term for that, because it's impossible to reproduce it now, since only `fold` has an access to iterable and it can't be spread to `call`. But I added just incase for possible future changes